plugins/outputs/cloudrun/cloudrun.go

```diff
@@ -0,0 +1,202 @@
+package cloudrun
+
+import (
+	"bytes"
+	"context"
+	"fmt"
+	"io"
+	"io/ioutil"
+	"net/http"
+	"time"
+
+	jwtGo "github.com/golang-jwt/jwt/v4"
+
+	"golang.org/x/oauth2/google"
+	"golang.org/x/oauth2/jwt"
+
+	"github.com/influxdata/telegraf"
+	"github.com/influxdata/telegraf/internal"
+
+	httpconfig "github.com/influxdata/telegraf/plugins/common/http"
+	"github.com/influxdata/telegraf/plugins/outputs"
+	"github.com/influxdata/telegraf/plugins/serializers"
+)
+
+const sampleConfig = `
+  ## A plugin that can transmit metrics over OAuth2
+  ## URL is the Cloud Run Wavefront proxy address to send metrics to
+  # url = "http://127.0.0.1:8080/telegraf"
+
+  ## Timeout for Cloud Run message, suggested as 30s to account for handshaking
+  # timeout = "30s"
+
+  ## Cloud Run JSON file location
+  ## This is the location of the JSON file generated from your GCP project that's authorized to send
+  ## metrics into CloudRun.
+  ## Windows users, note that you need to use forward slashes.
+  # credentials_file = "/etc/telegraf/example_secret.json"
+
+  ## Data format to output.
+  ## Each data format has it's own unique set of configuration options, read
+  ## more about them here:
+  ## https://github.com/influxdata/telegraf/blob/master/docs/DATA_FORMATS_OUTPUT.md
+  # data_format = "wavefront"
+
+  ## NOTE: The default headers have already been set to the following by default:
+  ## defaultContentType   = "application/octet-stream"
+  ## defaultAccept        = "application/json"
+  ## defaultMethod        = http.MethodPost
+`
+
+const (
+// defaultClientTimeout = 5 * time.Second
+// defaultContentType   = "application/octet-stream"
+// defaultAccept        = "application/json"
+// defaultMethod = http.MethodPost
+)
```

Comment:
Please remove this dead code.

Reply:
@srebhan Will remove the code

Yes, this plugin is meant to handle negotiating the two legged OAuth2 flow so as long as your data is in the expected format it should work as in the end it's just a REST request with a token and payload.

Reply:
We'll tackle your suggestion to create a CreateCloudRunClient function in common/oauth. 🙂